### PR TITLE
Use a command to build index.js and index.d.ts

### PR DESCRIPTION
**Pull request by BrandonKerr**
When Adrian demoed this package and explained the step of adding a new function declaration to both index.js and index.d.ts, it made my spidey sense tingle. I originally mistook the situation and thought of generating an index.js file from an index.ts file (not just the declaration .d.ts file). After realizing my mistake, I still saw some room for improvement to automate that process and avoid manual errors. 

While doing the structuring, I alphabetized the entries to make it easier to read, but it makes this diff even harder to really compare. I ordered the original import and export statements to compare and left comments on each file where there's a real difference. 

To generate the files, simply run `npm run build-index`

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -1,91 +1,140 @@
...
-    fetchAllPacks,
-    fetchPackAll,
-    fetchPackChildren,
-    fetchLessonContent,
-    fetchCourseOverview,
-    fetchChallengeOverview,
-    fetchScheduledReleases,
-    fetchArtistLessons,
-    fetchGenreLessons,
-  } from './services/sanity.js';
```
Comment (BrandonKerr) [inline src/index.d.ts:34]: This adds entries that are exported in sanity.js but were missing before:

- fetchByReference,
- fetchCatalogMetadata,
- fetchChildren,
- fetchCoachLessons,
- fetchMethodPreviousNextLesson,
- fetchParentByRailContentId,
- fetchRelatedMethodLessons,
- fetchSanity,
- getSortOrder

File: src/index.d.ts [hunk 2]
```diff
@@ -1,91 +1,140 @@
...
+	fetchCompletedState,
+	fetchContentInProgress,
+	fetchContentPageUserData,
+	fetchHandler,
+	fetchLikeContent,
+	fetchSongsInProgress,
+	fetchUnlikeContent,
+	fetchUserContext,
+	fetchVimeoData
 } from './services/railcontent.js';
```
Comment (BrandonKerr) [inline src/index.d.ts:20]: This adds entries that are exported in railcontent.js but were missing before:

- fetchHandler,
- fetchLikeContent,
- fetchSongsInProgress,
- fetchUnlikeContent,
- fetchUserContext,

File: src/index.d.ts [hunk 3]
```diff
@@ -1,91 +1,140 @@
...
+	fetchSanity,
+	fetchScheduledReleases,
+	fetchSongArtistCount,
+	fetchSongById,
+	fetchSongCount,
+	fetchSongFilterOptions,
+	fetchUpcomingEvents,
+	fetchWorkouts,
+	getSortOrder
+} from './services/sanity.js';
```
Comment (BrandonKerr) [inline src/index.d.ts:64]: This adds entries that are exported in sanity.js but were missing before:

- fetchCatalogMetadata,
- fetchSanity,
- getSortOrder

File: src/index.d.ts [hunk 4]
```diff
@@ -1,91 +1,140 @@
...
 
+import {
+	clearCache,
+	fetchContentData,
+	init,
+	likeContent,
+	testClearLocal,
+	unlikeContent,
+	version
+} from './services/userContext.js';
```
Comment (BrandonKerr) [inline src/index.d.ts:74]: userContext.js was not included before

File: src/index.js [hunk 5]
```diff
@@ -1,100 +1,138 @@
-import { initializeService } from './services/config.js';
```
Comment (BrandonKerr) [inline src/index.js:1]: Adds the globalConfig

File: src/index.js [hunk 6]
```diff
@@ -1,100 +1,138 @@
...
-  fetchMethodPreviousNextLesson,
-  fetchLiveEvent,
-  fetchChallengeOverview,
-  fetchCoachLessons,
-  fetchByReference,
-  fetchScheduledReleases,
-  fetchArtistLessons,
-  fetchGenreLessons,
-  fetchRelatedMethodLessons,
-} from './services/sanity.js';
```
Comment (BrandonKerr) [inline src/index.js:42]: This adds entries that are exported in sanity.js but were missing before:

- fetchCatalogMetadata,
- fetchSanity,
- getSortOrder

File: src/index.js [hunk 7]
```diff
@@ -1,100 +1,138 @@
...
+	fetchCompletedState,
+	fetchContentInProgress,
+	fetchContentPageUserData,
+	fetchHandler,
+	fetchLikeContent,
+	fetchSongsInProgress,
+	fetchUnlikeContent,
+	fetchUserContext,
+	fetchVimeoData
 } from './services/railcontent.js';
```
Comment (BrandonKerr) [inline src/index.js:20]: This adds entries that are exported in railcontent.js but were missing before:

- fetchHandler,
- fetchLikeContent,
- fetchSongsInProgress,
- fetchUnlikeContent,
- fetchUserContext,

File: src/index.js [hunk 8]
```diff
@@ -1,100 +1,138 @@
...
+
+import {
+	clearCache,
+	fetchContentData,
+	init,
+	likeContent,
+	testClearLocal,
+	unlikeContent,
+	version
+} from './services/userContext.js';
```
Comment (BrandonKerr) [inline src/index.js:74]: userContext.js was not included before

File: src/index.d.ts (hunk 1 above)
Comment (BrandonKerr) [inline src/index.d.ts:34]: This adds entries that are exported in sanity.js but were missing before:

- fetchByReference,
- fetchCatalogMetadata,
- fetchChildren,
- fetchCoachLessons,
- fetchMethodPreviousNextLesson,
- fetchParentByRailContentId,
- fetchSanity,
- getSortOrder

File: src/index.d.ts [hunk 9]
```diff
@@ -1,91 +1,140 @@
...
 
 declare module 'musora-content-services' {
-    export {
```
Comment (BrandonKerr) [inline src/index.d.ts:49]: Adds these exports:

- clearCache,
- fetchByReference,
- fetchCatalogMetadata,
- fetchChildren,
- fetchCoachLessons,
- fetchContentData,
- fetchHandler,
- fetchLikeContent,
- fetchMethodPreviousNextLesson,
- fetchParentByRailContentId,
- fetchRelatedMethodLessons,
- fetchSanity,
- fetchSongsInProgress,
- fetchUnlikeContent,
- fetchUserContext,
- getSortOrder,
- globalConfig,
- init,
- likeContent,
- testClearLocal,
- unlikeContent,
- version,

File: src/index.js [hunk 10]
```diff
@@ -1,100 +1,138 @@
...
 
 export {
```
Comment (BrandonKerr) [inline src/index.js:76]: Adds these exports:

- clearCache,
- fetchCatalogMetadata,
- fetchContentData,
- fetchContentData,
- fetchHandler,
- fetchLikeContent,
- fetchSanity,
- fetchSongsInProgress,
- fetchUnlikeContent,
- fetchUserContext,
- getSortOrder,
- globalConfig,
- init,
- likeContent,
- testClearLocal,
- unlikeContent,
- version,

**Review by adrian-musora — APPROVED** [review]

**Pull request merged**